docs/user/reporting/network-policy.asciidoc

```diff
@@ -0,0 +1,74 @@
+[role="xpack"]
+[[reporting-network-policy]]
+=== Restricting requests with a Reporting Network Policy
+
+When {report-features} generates PDF reports, it uses the Chromium browser to fully load the {kib} page on the server. This
+potentially involves sending requests to external hosts, for example, a request may go to an external image server for showing a
+field formatted as an image, or to show an image in a Markdown visualization.
+
+If the Chromium browser is requested to send a request that violates the network policy, Reporting will stop processing the page
+before the request goes out, and the report will be marked as a failure. Additional information about the event can be found in
+Kibana's server logs.
+
+A network policy applies not only to outgoing requests, but also incoming responses. That means if a request goes out to an allowed
+host, but is redirected and a response returns from a denied host, the response will be denied, and the report will fail.
+
+[NOTE]
+============
+{kib} installations are not designed to be publicly accessible over the Internet. The Reporting network policy and other capabilities
+of the Elastic Stack security features do not change this condition.
+============
+
+==== Configuring Reporting Network Policy
+
+You configure the network policy by specifying the `xpack.reporting.capture.networkPolicy.rules` setting in `kibana.yml`. A policy is specified as
+an array of objects that describe what to allow or deny based on an optionally-provided host and/or protocol. If a host or protocol
+is not specified, the rule will match any host or protocol, respectively.
+
+The rule objects are evaluated sequentially from the beginning to the end of the array, and continue until there is a matching rule.
+If none of the rules allow a request, the request will be denied.
+
+[source,yaml]
+-------------------------------------------------------
+# Only allow requests to placeholder.com
+xpack.reporting.capture.networkPolicy:
+  rules: [ { allow: true, host: "placeholder.com" } ] 
+-------------------------------------------------------
+
+[source,yaml]
+-------------------------------------------------------
+# Only allow requests to https://placeholder.com 
+xpack.reporting.capture.networkPolicy:
+  rules: [ { allow: true, host: "placeholder.com", protocol: "https:" } ] 
```

Comment:
@joelgriffith do you happen to know if the protocol string must always end with `:` ? I see it that way in the unit tests, but haven't delved deeper.

If that is the case, we should add schema validation to ensure it gets entered right.

Reply:
Yup, it should, likely an implementation bleed in how we parse the request. Schema validation would be great to add, IIRC this was prior to np, so I wonder if there's a better way to do so now

Reply:
https://github.com/elastic/kibana/pull/80766